---
title: "Lung Adenocarcinoma Analysis (TCGA, GDC)"
author:
- Artur Gomes | PG55692
- Catarina Gomes | PG55694
- Maria Carvalho | PG55130
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
  pdf_document: default
  word_document: default
---
<img src="EEUMLOGO.png" alt="Logo da Empresa" style="float:right; margin: 10px 10px 0 0; width: 110px; height: auto;" />

# {.tabset} 


## Dataset Download and Import

This analysis is conducted in R using Bioconductor packages and other essential libraries for statistical and machine learning applications. 

In this section, we will perform the following tasks: install and load the required packages, query and download gene expression data using the GDC data portal, and import the data into a suitable structure for downstream analysis.

### Installation and Import of Packages

This section presents all the packages used in this work, which support data acquisition and analysis throughout the document, enhancing the efficiency of retrieving and interpreting relevant information. The code checks whether each package is already installed and installs it if necessary.

```{r, results='hide', warning = FALSE, message=FALSE}

# Install BiocManager if not already installed
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")

# List of Bioconductor packages
bioc_packages <- c(
  "TCGAbiolinks",
  "DESeq2",
  "Biobase",
  "genefilter",
  "pheatmap",
  "org.Hs.eg.db",
  "fgsea",
  "SummarizedExperiment",
  "Glimma",
  "edgeR",
  "AnnotationDbi"
)

# Install missing Bioconductor packages
for (pkg in bioc_packages) {
  if (!requireNamespace(pkg, quietly = TRUE)) {
    BiocManager::install(pkg, ask = FALSE)
  }
}

# List of CRAN packages
cran_packages <- c(
  "jpeg",
  "ggbeeswarm",
  "ggplot2",
  "xfun",
  "factoextra",
  "Rtsne",
  "gplots",
  "RColorBrewer",
  "tidyverse",
  "rpart",
  "rsample",
  "caret"
)

# Install missing CRAN packages
for (pkg in cran_packages) {
  if (!requireNamespace(pkg, quietly = TRUE)) {
    install.packages(pkg)
  }
}

```

### Load Packages  

After installing the required packages, we need to load them into our R environment. The following command ensures all libraries are available for analysis:

```{r,results='hide', warning = FALSE, message=FALSE}
library(BiocManager)
library(jpeg)
library(TCGAbiolinks)
library(Biobase)
library(DESeq2)
library(ggbeeswarm)
library(genefilter)
library(pheatmap)
library(org.Hs.eg.db)
library(fgsea)
library(ggplot2)
library(xfun)
library(SummarizedExperiment) 
library(factoextra)
library(Rtsne)
library(gplots)
library(edgeR)
library(AnnotationDbi)
library(RColorBrewer)
library(tidyverse)
library(rpart)
library(rsample)
library(caret)
```

### Data Query and Preparation

The **`GDCquery()`** function is used to perform a query on the database. In this function, several parameters are defined to filter the dataset that will later be downloaded.

The **`project`** parameter corresponds to the ID assigned to the cancer type, allowing the selection of information specifically related to endometrial carcinoma of the uterine body. The **`data.category`** parameter is used to specify the data category to be selected. In this case, since the goal is to perform transcriptomic analysis, the category **"Transcriptome Profiling"** was selected to retrieve gene expression profiling data.

Among the available gene expression profiling data, the analysis was restricted to gene expression quantification by setting **`data.type`** to **"Gene Expression Quantification"**. Finally, the desired type of gene expression analysis was specified, opting for **"STAR-Counts"**. This workflow indicates that the algorithm used to count gene expression from aligned sequences was based on the **STAR** tool.

The **`GDCdownload()`** function is then used to download the data according to the query defined in the previous step. Next, the **`GDCprepare()`** function was used to generate a **`SummarizedExperiment`** object, compiling all the downloaded data into a single object to facilitate further analysis. It is worth noting that this function also allows saving the data in **.RData** format, making it easier to reload using the **`load()`** function.

> **Note**: This workflow enables the simultaneous download of both gene expression data and the associated metadata for the samples and genes.

```{r, message = FALSE}
# Define the base directory
base_dir <- "/home/arog/Documents/2_semestre/ECDB/project_data"  # Change this to your desired path

# Create directory if it doesn't exist
if (!dir.exists(base_dir)) {
  dir.create(base_dir, recursive = TRUE)
}

# File path for storing processed data
gdc_data <- file.path(base_dir, "GDCdata")
rna_seq_file <- file.path(base_dir, "mRNA_TCGA-LUAD.rda")

# Check if the data file already exists, if not, download and process
if (!file.exists(rna_seq_file)) {

  query_LUAD <- GDCquery(
    project = "TCGA-LUAD",
    data.category = "Transcriptome Profiling",
    data.type = "Gene Expression Quantification",
    workflow.type = "STAR - Counts"
  )

  GDCdownload(query = query_LUAD, directory = gdc_data)

  rna_seq_LUAD <- GDCprepare(
    query = query_LUAD,
    save = TRUE,
    save.filename = rna_seq_file,
    directory = gdc_data
  )

} else {
  rna_seq_LUAD <- get(load(rna_seq_file))
}

```


## Contextualization of the Data

The **TCGA-LUAD (The Cancer Genome Atlas - Lung Adenocarcinoma) project** is part of the Genomic Data Commons (GDC), aimed at characterizing lung adenocarcinoma at the molecular level. It provides **clinical, genomic, and molecular** data, including patient demographics, tumor characteristics, sequencing data, and protein expression profiles.  

This study focuses on analyzing **data and metadata** to assess quality, completeness, and usability. By examining metadata, we evaluate its role in ensuring data consistency, accessibility, and interoperability for research applications.  


### Objective

This project aims to:

- Identify differentially expressed genes (DEGs) in LUAD
- Investigate associations between gene expression and clinical variables
- Discover biomarkers or gene signatures for prognosis or treatment


## Initial Data Exploration
```{r}
cat("Dimensions:", dim(rna_seq_LUAD)[1], "genes x", dim(rna_seq_LUAD)[2], "samples\n")
```

### Gene and Sample Metadata

```{r}
gene_metadata <- rowData(rna_seq_LUAD)
sample_metadata <- colData(rna_seq_LUAD)
cat("Gene metadata:", dim(gene_metadata), "\n")
cat("Sample metadata:", dim(sample_metadata), "\n")
```


## Clinical Metadata Analysis

### Vital Status

```{r}
vital_status <- sample_metadata$vital_status
table_vs <- prop.table(table(vital_status)) * 100
barplot(table_vs, col = c("skyblue", "tomato"),
        main = "Vital Status of LUAD Patients", ylab = "%")
```

**Interpretation**: Majority of patients were [interpret here: alive or deceased?]. Vital status is useful for survival analysis.


### Tumor Stage (AJCC)

```{r}
stage <- sample_metadata$ajcc_pathologic_stage
table_stage <- sort(prop.table(table(stage)) * 100, decreasing = TRUE)
pie(table_stage, labels = paste(names(table_stage), sprintf("%.1f%%", table_stage)),
    col = brewer.pal(length(table_stage), "Pastel1"),
    main = "Tumor Stage Distribution")
```

**Interpretation**: Early-stage cases dominate, suggesting detection before extensive spread. Later stages correlate with poorer prognosis.


### Smoking History

```{r}
smoking <- sample_metadata$tobacco_smoking_history
table_smoke <- sort(prop.table(table(smoking, useNA = "ifany")) * 100, decreasing = TRUE)
pie(table_smoke, labels = paste(names(table_smoke), sprintf("%.1f%%", table_smoke)),
    col = brewer.pal(length(table_smoke), "Set3"),
    main = "Smoking History")
```

**Interpretation**: A large percentage of LUAD patients have a history of smoking. Smoking is a major LUAD risk factor.


### Age at Diagnosis

```{r}
age <- sample_metadata$age_at_index
summary(age)
boxplot(age, main = "Age at Diagnosis", col = "lightblue")
hist(age, main = "Age Distribution", xlab = "Age", col = "orange")
```

**Interpretation**: LUAD affects primarily older adults, typically diagnosed between 55 and 80.


## Data Preprocessing

```{r}
expression_data <- as.data.frame(assay(rna_seq_LUAD))
clinical_data <- colData(rna_seq_LUAD)
```

### Check Missing Values

```{r}
na_clinical <- colSums(is.na(clinical_data))
table_na <- data.frame(Variable = names(na_clinical), NAs = na_clinical)
table_na <- table_na[order(-table_na$NAs), ]
kableExtra::kable(table_na, caption = "Missing Values in Clinical Data")
```

Remove variables with >30% missing data:

```{r}
thresh <- 30
na_pct <- na_clinical / nrow(clinical_data) * 100
clinical_data_filtered <- clinical_data[, na_pct <= thresh]
```

### Filter Genes with Low Counts

```{r}
keep_genes <- rowSums(expression_data > 1) > ncol(expression_data) * 0.5
expression_filtered <- expression_data[keep_genes, ]
```


## Exploratory Gene Expression Analysis

### Sample Clustering Heatmap

```{r}
vsd <- vst(DESeqDataSetFromMatrix(countData = expression_filtered,
                                   colData = clinical_data_filtered,
                                   design = ~ 1))
mat <- assay(vsd)[1:50, ]  # Top 50 genes only for visualization
pheatmap(mat, scale = "row", show_rownames = FALSE,
         main = "Heatmap of Top 50 Genes (VST Transformed)")
```

### PCA

```{r}
pca <- prcomp(t(mat))
autoplot(pca, data = as.data.frame(colData(vsd)), colour = 'vital_status') +
  ggtitle("PCA of LUAD Samples")
```




